torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -484,6 +513,59 @@ void AliasDb::analyzeChunk(Node* node) {
   }
 }
 
+void AliasDb::analyzeWait(Node* node) {
+  // Do nothing, this node should have already been analyzed by the
+  // corresponding analyzeFork(). See that function for details.
+}
+
+void AliasDb::analyzeFork(Node* node) {
+  const auto subgraph = node->g(attr::Subgraph).get();
+  subgraphToOwner_.insert({subgraph, node});
+
+  const auto subgraphBlock = subgraph->block();
+  mapAliases(subgraphBlock->inputs(), node->inputs());
+  analyze(subgraphBlock);
+
+  // Propagate aliasing and write information from the subgraph outputs to the
+  // outputs of the corresponding aten::wait() calls, since that's where the
+  // values will eventually emerge.
+  const auto fut = node->output();
+  const auto subgraphWrites = getWrites(subgraph->block());
+  for (const auto& use : fut->uses()) {
+    const auto wait = use.user;
+    AT_ASSERT(wait->outputs().size() == subgraph->outputs().size());
```

Comment:
Will this fail is it's used in an if node ?

Reply:
Hmm…good point. I guess it's non-trivial to find all the matching "waits".